modules/applications/pages/how-to-handle-pagination.adoc

```diff
@@ -0,0 +1,64 @@
+= How to handle server-side pagination
+:description: Learn how to handle server-side pagination and optimize performance
+
+{description}
+
+* Difficulty: beginner
+* Prerequisites: have your interface xref:interact-with-your-bonita-process.adoc[interact] with Bonita process
+* Estimated time: 10 minutes
+
+
+If you’re using list widgets and if your Bonita API requests return a lot of data, you might want to rely on server side pagination to improve performance.
+Follow this 3-steps tutorial to know how.
+
+Let's create an interface where a user types something in an input widget, and then it filters the API’s results and automatically adapts the pagination of a list widget.
+
+  
+== 1. Create a JS object
+
+Go to the `JS` tab (between `Queries` and `UI`), create a new JS object called “Pagination” and copy paste the following code:
+
+[source, JS]
+----
+export default {
+	maxResults (headers) {
+		return parseInt( headers['Content-Range'][0].split('/')[1] );
+	}
+}
+----
+
+INFO: All Bonita API calls returning a collection of elements are paginated and their response contains a https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range[Content-Range header].
+
+== 2. Create a dynamic API request
+
+Go to the `Queries` tab and create a request that will adapt the pagination depending on what is typed in an input widget:
+
+* `Name`: getUsers
+* `Method`: GET
+* `URL`: /bonita/API/identity/user
+* `Params`:
+    - `Key`: p
+    - `Value`: {{(List1.pageNo-1)}}
+    - `Key`: c
+    - `Value`: {{List1.pageSize}}
+    - `Key`: s
+    - `Value`: {{Input1.text}}
+
+
+== 3. Configure the interface
+
+On the interface, drag and drop an input widget (Input1) and a list widget (List1).
+
+Configure the input widget:
```

Comment:
This can be removed. Input widget is used for filtering the list which is not part of this guide